tests/TestCompiler.hs

```diff
@@ -55,55 +55,45 @@ import Text.Regex.TDFA (Regex)
 import TestUtils
 import Test.Hspec
 
-spec :: Spec
+spec :: SpecWith ([P.Module], [P.ExternsFile], M.Map P.ModuleName FilePath)
 spec = do
-  (supportModules, supportExterns, supportForeigns) <- runIO setupSupportModules
+  passingTests
+  warningTests
+  failingTests
 
-  passingTests supportModules supportExterns supportForeigns
-  warningTests supportModules supportExterns supportForeigns
-  failingTests supportModules supportExterns supportForeigns
-
-passingTests
-  :: [P.Module]
-  -> [P.ExternsFile]
-  -> M.Map P.ModuleName FilePath
-  -> Spec
-passingTests supportModules supportExterns supportForeigns = do
+passingTests :: SpecWith ([P.Module], [P.ExternsFile], M.Map P.ModuleName FilePath)
+passingTests = do
   passingTestCases <- runIO $ getTestFiles "passing"
-  outputFile <- runIO $ createOutputFile logfile
 
   describe "Passing examples" $
-    forM_ passingTestCases $ \testPurs ->
-      it ("'" <> takeFileName (getTestMain testPurs) <> "' should compile and run without error") $
-        assertCompiles supportModules supportExterns supportForeigns testPurs outputFile
-
-warningTests
-  :: [P.Module]
-  -> [P.ExternsFile]
-  -> M.Map P.ModuleName FilePath
-  -> Spec
-warningTests supportModules supportExterns supportForeigns = do
+    beforeAllWith ((<$> createOutputFile logfile) . (,)) $
```

Comment:
This probably shouldn't happen in this PR but I'd also like to consider getting rid of this logfile mechanism. I have never used it and I'm not sure anyone else is using it either, and I think it's easy enough to run `stack exec bash` or something and look at the compiler output, and also I think the need for it was largely supplanted by the golden tests on the diagnostics.